### DISCUSSION TITLE
Ask your question here.

### DISCUSSION THREAD

**Discussion by Keyitdev** [Deerrorer/dotfiles3#5] [Q&A]
If you have any questions, ask them here.

**Comment by alphaprads** [comment]
which file is responsible for login manager behaviour in short the login manager layout and background.

**Reply by Keyitdev** [reply]
https://github.com/Keyitdev/sddm-flower-theme/tree/master/Components

https://github.com/Keyitdev/sddm-flower-theme/blob/master/theme.conf

`/usr/share/sddm/themes/sddm-flower-theme`

Those files are the most important. But the whole repository is responsible for appearance and behavior.

**Reply by alphaprads** [reply]
oh thanks

**Comment by alphaprads** [comment]
hello i wanted to work on this project as it is something i use daily i saw that the battery icon isn't dynamic so i thought of working on it and i could make it work dynamically are you alright with me making a pull request this i my first contribution to non local repo on github so i thought of asking

**Comment by PedroPepeu** [comment]
The issue of cannot login, was already solved?

**Reply by Keyitdev** [reply]
I think so.

https://github.com/Keyitdev/dotfiles/issues/10

It's hard to tell what's wrong.
If you can't log in, it could be due to many things.

You can try to skip SDDM

https://github.com/Keyitdev/dotfiles/issues/10#issuecomment-1562507180

**Reply by PedroPepeu** [reply]
Sorry for the time I keep you waiting, I reinstall the WM but I did not add the SDDM theme, so it's working, if I try it again and see that is better I comeback here to close the thread, thanks for the attention

**Reply by Keyitdev** [reply]
Hey, I actually solved the problem. It wasn't logging because i3wm was not installed. The light package was removed from the official repository. This caused problems with the installation of the other packages.
So if something is still broken, I recommend installing it manually.

Soon I will provide a new version of the installation script. (I hope)
Cheers

**Comment by Arya-LL** [comment]
Hi, how can I change the opacity of inactive windows? I am trying to make inactive windows less transparent. Thanks for the dotfiles, these are incredible!

**Comment by rustedusted** [comment]
i have a list of few things i would like to ask btw 
if you're alright i can just list them here

- why did we use exec_always for light -I command
- also I have a udev file written if you want we can directly add that udev file which will reduce unnecessary shortcut keys of swapping between hdmi connections using mod+m and mod+n

btw the polybar config you've created is amazing btw but the newer version of polybar is giving few warnings about the way you've written it as few styles of writing the config seem to have been deprecated 


also i haven't used rofi before i don't know a lot about it can you tell me what are rofi-bookmarks and how to use it i mean how do i add bookmarks are those bookmarks to webpages or something else?

i don't know a lot i have to admit and would love to hear your thoughts on these things

**Reply by Keyitdev** [reply]
Hi, some parts of the dotfiles dosen't work. Light is deprecated by maintainer. Rofi bookmarks dosen't work correctly because firefox changed how it stores bookrmarks.

I am working on the v4 version, but currently I don't have much time.
Polybar is fixed in v4 version.
Light will be replaced.
And rofi-bookrmarsk will probably be discounted.

**Reply by rustedusted** [reply]
i would suggest using brightnessctl for brightness i've replaced the light commands with exactly that tbh

`# brightness control
exec_always --no-startup-id brightnessctl set 10%
bindsym XF86MonBrightnessUp exec brightnessctl set +5% && brightness &
bindsym XF86MonBrightnessDown exec brightnessctl set 5%- && brightness &
`

this works great for me btw 
btw what do you think about the udev stuff 
i am good at it if you want i can add udev stuff for hdmi and bluetooth as well

**Reply by Keyitdev** [reply]
In the v4 version  light will probably be replaced by my simple shell script: 
```sh
#!/bin/sh

# default
step=+5

video_card=$(ls -1 /sys/class/backlight/)
cur_brightness=$(cat /sys/class/backlight/"$video_card"/brightness)
max_brightness=$(cat /sys/class/backlight/"$video_card"/max_brightness)

while getopts s: flag
do
    case "${flag}" in
        s) step=${OPTARG};;
        *) echo "Usage: -s <-value/+value>";;
    esac

done

step=$(echo "$step" | sed 's/+//g')
step_c=$(echo "scale=2; $step*($max_brightness/100)" | bc -l | sed 's/\..*$//')
new_brightness=$((cur_brightness + step_c))

if [ "$new_brightness" -le "0" ]; then
    echo "0" > /sys/class/backlight/"$video_card"/brightness
elif [ "$new_brightness" -ge "$max_brightness" ] ; then
    echo "$max_brightness" > /sys/class/backlight/"$video_card"/brightness
else
   echo "$new_brightness" > /sys/class/backlight/"$video_card"/brightness
fi
```
I think udev is better for "plug and play" experience, but with keybindings you get more control.

If you want to help you can try to fix rofi-bookmarks. It requires basic knowledge of shell scripting and some sql. Can you do it?  

![image](https://github.com/user-attachments/assets/cf204919-080b-421d-80cc-3ef0de490f89)

**Comment by rustedusted** [comment]
btw the config file doesn't work well with bluetooth earphones for me is that just issue for me or does the dotfiles not handle external audio devices or is it issue of polybar ?

**Reply by Keyitdev** [reply]
Unfortunately the dotfiles do not nclude bluetooth control.

**Comment by matrixmilanila** [comment]
how can i add spotify to the music thing? i wanna show off my epic music taste

**Reply by Keyitdev** [reply]
Hi, my configs only includes music from MPD.  But if you want to add Spotify module, you can try something like this https://github.com/PrayagS/polybar-spotify.

**Comment by Gokits45** [comment]
hi, can i ask for an update? i think the install script and manually install tutorial are kinda outdated and dont work very well

**Reply by PedroPepeu** [reply]
Hey, they probably are, since the last commit has more than an year, are you new into ricing? If you need some help I am not the maintainer but I would be glad to help

**Reply by Gokits45** [reply]
yeah im new, if you could help that would be great, thx!

**Reply by PedroPepeu** [reply]
Just say what you need, I will try to do an PR to update the code, if there is something wrong, but say it here and I will try to help you. Also, if it helps to you feel more secure about the helps, or to be updated about ricing you can access [r/UnixPorn](https://www.reddit.com/r/unixporn/) it is an reddit community that talks about ricing and unix personalization.
